aspnetcore/signalr/configuration.md

```diff
@@ -325,6 +325,61 @@ var connection = new signalR.HubConnectionBuilder()
 
 ---
 
+### Configure stateful reconnect
+
+Stateful reconnect reduces the perceived downtime of clients that have a temporary disconnect in their network connection, such as when switching network connections or a short temporary loss in access. Stateful reconnect is available in ASP.NET Core 8.0 and later.
+
+Stateful reconnect achieves this by:
+
+* Temporarily buffering data on the server and client.
+* Acknowledging messages received (ACK-ing) by both the server and client.
+* Recognizing when a connection is returning and replaying messages that may have been sent while the connection was down.
+
+To opt-in to stateful reconnect for a JavaScript or Typescript client:
+
+* Update the JavaScript or TypeScript client code to enable the `withStatefulReconnect` option:
+
+  ```JavaScript
+  const builder = new signalR.HubConnectionBuilder()
+    .withUrl("/hubname")
+    .withStatefulReconnect({ bufferSize: 1000 });  // Optional, defaults to 100,000
+  const connection = builder.build();
+  ```
+
+  The `bufferSize` option is optional with a default of 100,000 bytes.
+
+* Update the server hub endpoint configuration to enable the `AllowStatefulReconnects` option:
+
+  ```csharp
+  app.MapHub<MyHub>("/hubName", options =>
+  {
+      options.AllowStatefulReconnects = true;
+  });
+  ```
+
+To opt-in to stateful reconnect for a .NET client:
+
+* Update the .NET client code to enable the `WithStatefulReconnect` option:
+
+  ```csharp
+    var builder = new HubConnectionBuilder()
+        .WithUrl("<hub url>")
+        .WithStatefulReconnect();
+    builder.Services.Configure<HubConnectionOptions>(o => o.StatefulReconnectBufferSize = 1000);
+    var hubConnection = builder.Build();
+  ```
+
+  The `StatefulReconnectBufferSize` option is optional with a default of 100,000 bytes.
+
+* Update the server hub endpoint configuration to enable the `AllowStatefulReconnects` option:
```

Comment:
I think this should be listed first. Server must allow stateful reconnect otherwise client settings are ignored.

We also don't need to put it for both clients, if it's at the top that should be obvious 😄